crates/net/discv5/src/lib.rs

```diff
@@ -541,18 +541,14 @@ pub fn get_lookup_target(
     let mut target = local_node_id.raw();
 
     // make sure target has a 'log2distance'-long suffix that differs from local node id
-    if kbucket_index != 0 {
-        let suffix_bit_offset = MAX_KBUCKET_INDEX.saturating_sub(kbucket_index);
-        let suffix_byte_offset = suffix_bit_offset / 8;
-        // todo: flip the precise bit
-        // let rel_suffix_bit_offset = suffix_bit_offset % 8;
-        target[suffix_byte_offset] = !target[suffix_byte_offset];
-
-        if suffix_byte_offset != 31 {
-            for b in target.iter_mut().take(31).skip(suffix_byte_offset + 1) {
-                *b = rand::random::<u8>();
-            }
-        }
+    let suffix_bit_offset = MAX_KBUCKET_INDEX.saturating_sub(kbucket_index);
+    let suffix_byte_offset = suffix_bit_offset / 8;
+    // todo: flip the precise bit
+    // let rel_suffix_bit_offset = suffix_bit_offset % 8;
+    target[suffix_byte_offset] = !target[suffix_byte_offset];
+
+    for b in target.iter_mut().skip(suffix_byte_offset + 1) {
+        *b = rand::random::<u8>();
     }
```

Comment:
```suggestion
    rand::thread_rng().fill_bytes(&mut target[suffix_byte_offset..]);
```

Reply:
wouldn't the slice start at `suffix_byte_offset + 1`?

Reply:
yep according to the previous code, typo

Reply:
was also wrong in the issue saw now